tests/test_other.py

```diff
@@ -9657,29 +9657,46 @@ def test_no_excessive_invoke_functions_are_generated_when_exceptions_are_enabled
 
   def test_add_emscripten_metadata(self):
     run_process([PYTHON, EMCC, path_from_root('tests', 'hello_world.c'),
-                 '-s', 'EMIT_EMSCRIPTEN_METADATA',
-                 '-o', 'hello_world.js'])
-    wasm = open('hello_world.wasm', 'rb').read()
+                 '-s', 'EMIT_EMSCRIPTEN_METADATA'])
+    wasm = open('a.out.wasm', 'rb').read()
     # emscripten_metadata should be in the wasm data
     offset = 8 # skip magic + header
-    for _ in range(100):
+    while True:
       section = wasm[offset:offset + 1]
-      self.assertEqual(section, b'\0', 'No emscripten_metadata section found before standard wasm sections')
+      # No emscripten_metadata section found before standard wasm sections')
+      self.assertEqual(section, b'\0')
       offset += 1
-      (section_size, offset) = WebAssembly.delebify(wasm, offset)
+      section_size, offset = WebAssembly.readLEB(wasm, offset)
       end_offset = offset + section_size
-      (name_len, offset) = WebAssembly.delebify(wasm, offset)
+      name_len, offset = WebAssembly.readLEB(wasm, offset)
       name = wasm[offset:offset + name_len]
       if name == b'emscripten_metadata':
         break
       offset = end_offset
-    else:
-      self.assertFalse("No emscripten_metadata section found in first 100 custom sections")
+      if offset >= len(wasm):
+        self.assertFalse('No emscripten_metadata section found')
 
     # make sure wasm executes correctly
-    ret = run_process(NODE_JS + ['hello_world.js'], stdout=PIPE).stdout
+    ret = run_process(NODE_JS + ['a.out.js'], stdout=PIPE).stdout
     self.assertTextDataIdentical('hello, world!\n', ret)
 
+  def test_no_emscripten_metadata(self):
+    run_process([PYTHON, EMCC, path_from_root('tests', 'hello_world.c')])
+    wasm = open('a.out.wasm', 'rb').read()
+    # emscripten_metadata should be in the wasm data
+    offset = 8 # skip magic + header
+    while offset < len(wasm):
+      section = wasm[offset:offset + 1]
+      offset += 1
+      section_size, offset = WebAssembly.readLEB(wasm, offset)
+      end_offset = offset + section_size
+      # if this is a custom section
+      if section == b'\0':
+        name_len, offset = WebAssembly.readLEB(wasm, offset)
+        name = wasm[offset:offset + name_len]
+        self.assertNotEqual(name, b'emscripten_metadata')
```

Comment:
how about just asserting this string is nowhere in the entire wasm binary? not sure why the loop and all the parsing was in the original version...